src/Makefile.inc

```diff
@@ -0,0 +1,340 @@
+
+# We want to turn on INCLUDE_FS by default
+INCLUDE_FS ?= 1
+
+# Use VERBOSE=1 to enable verbose make
+PROGMODE ?= STATUS_SCREEN
+
+NUM_MODELS ?= 30
+TYPE     ?= prd
+
+###############################################
+#This section defines binaries needed to build#
+###############################################
+ifdef CROSS
+    CC   = $(CROSS)gcc
+    CXX  = $(CROSS)g++
+    LD   = $(CROSS)ld
+    AR   = $(CROSS)ar
+    AS   = $(CROSS)as
+    CP   = $(CROSS)objcopy
+    DUMP = $(CROSS)objdump
+    NM   = $(CROSS)nm
+endif
+CC   ?= gcc
+CXX  ?= g++
+LD   ?= ld
+AR   ?= ar
+AS   ?= as
+CP   ?= objcopy
+DUMP ?= objdump
+###############################################
+#END SECTION                                  #
+###############################################
+
+############################################
+#This section defines the source files used#
+############################################
+#ifndef SCREENSIZE
+#error please define SCREENSIZE
+#endif
+
+SRC_C    +=  $(wildcard $(SDIR)/firmware/*.c) \
+             $(wildcard $(SDIR)/misc/*.c) \
+             $(wildcard $(SDIR)/screen/*.c) \
+             $(wildcard $(SDIR)/screen/$(SCREENSIZE)/*.c) \
+             $(wildcard $(SDIR)/gui/*.c) \
+             $(wildcard $(SDIR)/pages/$(SCREENSIZE)/*.c) \
+             $(wildcard $(SDIR)/config/*.c)
+
+ifdef MODULAR
+   SRC_C    += $(SDIR)/protocol/protocol.c
```

Comment:
I never liked this.  Modular builds are a target thing, and don't belong in the main makefile.  but it is weird because the protocol code itself is here.  I'm not sure what to do about it.